sdk/servicebus/azure-servicebus/azure/servicebus/_common/utils.py

```diff
@@ -9,20 +9,56 @@
 import logging
 import threading
 import time
+import functools
 try:
     from urlparse import urlparse
 except ImportError:
     from urllib.parse import urlparse
 from concurrent.futures import ThreadPoolExecutor
 
-from uamqp.constants import TransportType
+from uamqp import authentication
 
-from azure.servicebus.common.errors import AutoLockRenewFailed, AutoLockRenewTimeout
-from azure.servicebus import __version__ as sdk_version
+from ..exceptions import AutoLockRenewFailed, AutoLockRenewTimeout
+from .._version import VERSION as sdk_version
+from .constants import (
+    JWT_TOKEN_SCOPE
+)
 
 _log = logging.getLogger(__name__)
 
 
+class UTC(datetime.tzinfo):
+    """Time Zone info for handling UTC"""
+
+    def utcoffset(self, dt):
+        """UTF offset for UTC is 0."""
+        return datetime.timedelta(0)
+
+    def tzname(self, dt):
+        """Timestamp representation."""
+        return "Z"
+
+    def dst(self, dt):
+        """No daylight saving for UTC."""
+        return datetime.timedelta(hours=1)
+
+
+try:
+    from datetime import timezone  # pylint: disable=ungrouped-imports
+
+    TZ_UTC = timezone.utc  # type: ignore
+except ImportError:
+    TZ_UTC = UTC()  # type: ignore
+
+
+def utc_from_timestamp(timestamp):
+    return datetime.datetime.fromtimestamp(timestamp, tz=TZ_UTC)
```

Comment:
I'm always nervous to open my mouth when it comes to timezone issues, but this is going to cause datetime comparison issues with many common scenarios, e.g. utcfromtimestamp returning a tzinfo of None.  Unfortunately our guidelines don't offer principles for whether we should go timezone aware or not; and your intentional choice here makes me want to ask first if this is purposeful?

Reply:
yes, it's purposeful. I copied what EventHub is doing, I think @annatisch made this part.
At the beginning I'm wondering why this way, so I checked python doc -- 
https://docs.python.org/3/library/datetime.html#datetime.datetime.utcfromtimestamp

It says:
"""
Warning Because naive datetime objects are treated by many datetime methods as local times, it is preferred to use aware datetimes to represent times in UTC. As such, the recommended way to create an object representing a specific timestamp in UTC is by calling datetime.fromtimestamp(timestamp, tz=timezone.utc).
"""